Add RFC: Reverse Domain Name Notation naming convention

Xaymar opened the pull request:
### Description
Adopt the [Reverse Domain Name notation](https://en.wikipedia.org/wiki/Reverse_domain_name_notation) naming convention to reduce potential collisions.

### Motivation and Context
Naming for additions has no guarantee that future additions to OBS or plugins do not eventually collide and corrupt setups. A unified naming convention that by design already prevents this should be adopted and enforced.

### [Link to RFC](https://github.com/xaymar/obs-rfcs/blob/rfc3/text/0041-reverse-domain-name-notation.md)

Xaymar commented:
Partial change extracted from #34 and #35 in order to reduce massive RFCs into smaller ones.

WizardCM commented:
I personally like this sound of this change, but I feel a migration path / backwards compatibility would need to be fleshed out extensively. For example, somehow check what version of libobs a plugin is built against - and if it's old enough, ignore the notation rule but warn the user that the plugin may break/conflict.

Xaymar commented:
> @WizardCM: I feel a migration path / backwards compatibility would need to be fleshed out extensively.

I have added a possible migration path that should be compatible with old and new plugins.

> For example, somehow check what version of libobs a plugin is built against - and if it's old enough, ignore the notation rule but warn the user that the plugin may break/conflict.

Added, though it is considered an optional component.

Xaymar commented:
Closing as requested.
